moveit_core/collision_detection/include/moveit/collision_detection/collision_robot.h

```diff
@@ -59,7 +59,7 @@ class CollisionRobot
    *  @param padding The padding to use for all objects/links on the robot
    *  @scale scale A common scaling to use for all objects/links on the robot
    */
-  CollisionRobot(const robot_model::RobotModelConstPtr& model, double padding = 0.0, double scale = 1.0);
+  CollisionRobot(robot_model::RobotModelConstPtr model, double padding = 0.0, double scale = 1.0);
```

Comment:
Replacing const-ref by value argument is a suggestion of clang tidy??
This shouldn't be replaced.

Reply:
The reasons for the replacement are on this page: https://clang.llvm.org/extra/clang-tidy/checks/modernize-pass-by-value.html. Looks like it only does so for constructors at the moment.

Easy enough to turn off though, I will do so.

Reply:
Explanation makes sense in general. However, for shared ptrs, move-construction doesn't make sense. Usually you want to keep the pointer afterwards. So, no need to turn this feature off, but asking for manual validation of all changes (and reverting unnecessary ones).